src/AutoRest.CSharp/Common/Output/Models/Types/ModelFactoryTypeProvider.cs

```diff
@@ -40,6 +40,11 @@ private static bool RequiresModelFactory(SchemaObjectType model)
                 return false;
             }
 
+            if (model.Properties.Any(p => p.Declaration.Accessibility != "public" && p.SchemaProperty?.IsDiscriminator != true))
```

Comment:
Why the discriminator part?

Reply:
Discriminator property is always internal.